### PR TITLE
Oauth fixes

### LINKED ISSUE #157: Fix OAuth
[provide a description of the issue]
I am led to understand that OAuth is not working correctly in apicast at the moment.

Can anyone verify that?

##### Steps To Reproduce

1. [step 1 (json configuration file, if applies)]
2. [step 2 (curl commands to reproduce, if applies)]
3. [step 3]

##### Additional Information

* [Gist with minimal reproducible configuration, see guidelines for contributing for details]
* [Gist with nginx log output]

### PR DESCRIPTION
**Pull request by mpguerra**
Closes #157

### CODE REVIEW

File: t/005-apicast-oauth.t [hunk 1]
```diff
@@ -60,14 +60,15 @@ Location: http://example.com/redirect?error=invalid_client
...
 
   set $backend_endpoint 'http://127.0.0.1:$TEST_NGINX_SERVER_PORT/backend';
+  set $backend_host '127.0.0.1';
   set $service_id 42;
   set $backend_authentication_type 'provider_key';
   set $backend_authentication_value 'fookey';
 
   location = /backend/transactions/oauth_authorize.xml {
     content_by_lua_block {
       expected = "provider_key=fookey&service_id=42&redirect_uri=otheruri&app_id=id"
```
Comment (andrewdavidmackenzie) [inline t/005-apicast-oauth.t]: fookey..

need parameterization of this string here???

**Comment by davidor** [comment]
👍

**Comment by mikz** [comment]
This will need test coverage, because OAuth was not working at all, right?

**Comment by mikz** [comment]
Not missing test coverage. Good job 👍 
Deleting myself.

**Pull request merged**